aiobotocore/httpxsession.py

```diff
@@ -0,0 +1,305 @@
+        # must specify a default or set all four parameters explicitly
+        # 5 is httpx default default
+        self._timeout = httpx.Timeout(
+            5, connect=conn_timeout, read=read_timeout
+        )
+
+        self._cert_file = None
+        self._key_file = None
+        if isinstance(client_cert, str):
+            self._cert_file = client_cert
+        elif isinstance(client_cert, tuple):
+            self._cert_file, self._key_file = client_cert
+        elif client_cert is not None:
+            raise TypeError(f'{client_cert} must be str or tuple[str,str]')
+
+        # previous logic was: if no connector args, specify keepalive_expiry=12
+        # if any connector args, don't specify keepalive_expiry.
+        # That seems .. weird to me? I'd expect "specify keepalive_expiry if user doesn't"
+        # but keeping logic the same for now.
+        if connector_args is None:
+            # aiohttp default was 30
+            # AWS has a 20 second idle timeout:
+            #   https://web.archive.org/web/20150926192339/https://forums.aws.amazon.com/message.jspa?messageID=215367
+            # "httpx default timeout is 5s so set something reasonable here"
+            self._connector_args: dict[str, Any] = {'keepalive_timeout': 12}
+        else:
+            self._connector_args = connector_args
+
+        if 'use_dns_cache' in self._connector_args:
+            raise NotImplementedError(
+                "DNS caching is not implemented by httpx. https://github.com/encode/httpx/discussions/2211"
+            )
+        if 'force_close' in self._connector_args:
+            raise NotImplementedError("Not supported with httpx as backend.")
+        if 'resolver' in self._connector_args:
+            raise NotImplementedError("Not supported with httpx as backend.")
+
+        self._max_pool_connections = max_pool_connections
+        self._socket_options = socket_options
+        if socket_options is None:
+            self._socket_options = []
+
+        # TODO [httpx]: clean up
+        ssl_context: SSLContext | None = None
+        self._verify: bool | str | SSLContext = verify
+        if not verify:
+            return
+        if 'ssl_context' in self._connector_args:
+            self._verify = cast(
+                'SSLContext', self._connector_args['ssl_context']
+            )
+            return
+
+        ssl_context = self._get_ssl_context()
+
+        # inline self._setup_ssl_cert
+        ca_certs = get_cert_path(verify)
+        if ca_certs:
+            ssl_context.load_verify_locations(ca_certs, None, None)
+        if ssl_context is not None:
+            self._verify = ssl_context
+
+    async def __aenter__(self):
+        assert not self._session
+
+        limits = httpx.Limits(
+            max_connections=self._max_pool_connections,
+            # 5 is httpx default, specifying None is no limit
+            keepalive_expiry=self._connector_args.get('keepalive_timeout', 5),
```

Comment:
lets do this in one place with one default, there are now two defaults a 5 and a 12.  Just set it once in `__init__`